src/Uno.UI/UI/Xaml/UIElement.Pointers.Skia.cs

```diff
@@ -0,0 +1,328 @@
+
+				var pointer = new Pointer(args.CurrentPoint.PointerId, PointerDeviceType.Mouse, false, isInRange: true);
+
+				if (UIElement.PointerCapture.TryGet(pointer, out var capture))
+				{
+					foreach (var target in capture.Targets)
+					{
+						var pointerArgs = new PointerRoutedEventArgs(args, pointer, target.Element) { CanBubbleNatively = true };
+
+						TraverseAncestors(target.Element, pointerArgs, e => e.OnNativePointerMove(pointerArgs));
+					}
+				}
+				else
+				{
+					PropagateEvent(args, e =>
+					{
+						var pointerArgs = new PointerRoutedEventArgs(args, pointer, e) { CanBubbleNatively = true };
+						TraverseAncestors(e, pointerArgs, e2 => e2.OnNativePointerMove(pointerArgs));
+					});
+				}
+			}
+
+			private void TraverseAncestors(UIElement element, PointerRoutedEventArgs args, Action<UIElement> action)
+			{
+				action(element);
+
+				foreach(var parent in element.GetParents().OfType<UIElement>())
+				{
+					action(parent);
+
+					if (args.Handled)
+					{
+						break;
+					}
+				}
+			}
+
+			private void PropagateEvent(PointerEventArgs args, Action<UIElement> raiseEvent)
+			{
+				if(Window.Current.Content is UIElement root)
+				{
+					PropagageEventRecursive(args, new Point(0, 0), root, raiseEvent);
+				}
+			}
+
+			private bool PropagageEventRecursive(PointerEventArgs args, Point root, UIElement element, Action<UIElement> raiseEvent)
+			{
+				bool raised = false;
+				var rect = element.LayoutSlotWithMarginsAndAlignments;
+				rect.X += root.X;
+				rect.Y += root.Y;
+
+				var position = args.CurrentPoint.Position;
+
+				if (element.RenderTransform != null)
+				{
+					position = element.RenderTransform.Inverse.TransformPoint(position);
+				}
+
+				if (position.X >= rect.X	
+					&& position.Y >= rect.Y
+					&& position.X <= rect.X + rect.Width
+					&& position.Y <= rect.Y + rect.Height)
+				{
+
+					foreach (var e in element.GetChildren().ToArray())
+					{
+						raised |= PropagageEventRecursive(args, rect.Location, e, raiseEvent);
+					}
+
+					var isHitTestVisible =
+						element.GetValue(HitTestVisibilityProperty) is HitTestVisibility hitTestVisibility
+						&& hitTestVisibility == HitTestVisibility.Visible;
+
+					if (!raised && isHitTestVisible)
+					{
+						if (!element._pointerEntered)
+						{
+							// Console.WriteLine($"PointerManager.Entered [{element}/{element.GetHashCode():X8}");
+							element._pointerEntered = true;
+
+							var pointer = new Pointer(args.CurrentPoint.PointerId, PointerDeviceType.Mouse, false, isInRange: true);
+
+							var pointerArgs = new PointerRoutedEventArgs(args, pointer, element);
+							TraverseAncestors(element, pointerArgs, e => e.OnNativePointerEnter(pointerArgs));
+						}
+
+						raiseEvent(element);
+						raised = true;
+					}
+				}
+				else
+				{
+					if (element._pointerEntered)
+					{
+						element._pointerEntered = false;
+						// Console.WriteLine($"PointerManager.Exited [{element}/{element.GetHashCode():X8}");
```

Comment:
Codacy found an issue: [Remove this commented out code.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=5872302)